### DISCUSSION TITLE
Gameplay is -60 to -75 ms off

### DISCUSSION THREAD

**Discussion by ErtieNotReallyCodes** [ppy/osu#28600] [Q&A]
### Type

Game behaviour

### Bug description

Recently I started noticing that my offset was off. I tested it with an offset finder map and the hit error was -75 so I adjusted. But hitsounds sounded really off. I thought it was the audio, but with my offset, other things seemed off like the beat when you hover over the osu! logo, or the metronome in the editor. So, I concluded my **gameplay** is the one that's off. I don't know why. It says lazer on the version, but I also encountered this problem on stable.

### Version

Lazer 2024.521.2

### Logs

[compressed-logs.zip](https://github.com/user-attachments/files/15975291/compressed-logs.zip)

**Comment by Stoppedpuma** [comment]
This is an issue with your audio setup, if you are using bluetooth or some other form of wireless connection consider wired, connecting your output directly to your motherboard is the lowest audio latency solution. Using a program such as [REAL](https://github.com/miniant-git/REAL) can help further reduce this latency on windows.

**Reply by ErtieNotReallyCodes** [reply]
I said the gameplay was off. The audio is fine, because when I hover over the circle in the osu logo with 0 ms offset, it is on-beat.
With my offset, it is offbeat, but the gameplay is on-beat.

**Reply by Kori3** [reply]
That's literally what offset does, but i do see your point when talking about the menus.

As a suggestion it could be something like only applying offset for the gameplay, cuz everything else that doesn't require synced player inputs feels off with the sound effects.

about the hit sounds in game though I don't think that's fixable. If it annoys you too much I'd recommend muting the sound effects entirely or modifying your skin to have completely silent hit sounds

**Reply by Stoppedpuma** [reply]
> I said the gameplay was off. The audio is fine, because when I hover over the circle in the osu logo with 0 ms offset, it is on-beat.
With my offset, it is offbeat, but the gameplay is on-beat.

Sorry about that then, misread that part.

This could be a few different things then, hardware input delay (polling rate, input thread, etc),  beatmap offset, beatmap hitsounds, or if you're using a custom skin it might be the hitsounds in the skin itself. 

If you believe you're the one who's off beat then I don't know why you opened this in the first place? There's always this option when playing a beatmap though.

![image](https://github.com/ppy/osu/assets/58333920/6f43f648-8c05-4468-8bf6-8805653632cf)

**Reply by ErtieNotReallyCodes** [reply]
Look, I get what you guys mean, but I really wanna play with hitsounds. Maybe it would be cool if there would be a gameplay offset slider? Just saying. I'm still looking for a fix, though.

**Reply by Stoppedpuma** [reply]
That is a gameplay offset slider? it's per beatmap.

**Reply by Kori3** [reply]
> Maybe it would be cool if there would be a gameplay offset slider?

You can't do it for hitsounds, those play when you hit a note They're not embedded into the beatmap or something like that. 

The only viable workaround without messing up the experience would be turning sound effects completely off or playing with 0ms
